### ISSUE TITLE
Add rotate gesture support for images

### ISSUE THREAD

**Issue by jywarren** [publiclab/Leaflet.DistortableImage#165]
Using https://zingchart.github.io/zingtouch/ for cross browser support, perhaps?

**Comment by jywarren** [comment]
And for rotating groups, later, perhaps? @sashadev-sky this'll be cool!

**Comment by sashadev-sky** [comment]
@jywarren yes I will add this on my PR to do list for multiple image select! I got the basic functionality working this week - so you will see a PR for that today or tomorrow.

**Comment by jywarren** [comment]
oh this is so rad! Looking forward to it. Thanks!

**Comment by jywarren** [comment]
Noting https://stackoverflow.com/questions/13863974/rotate-element-based-on-touch-event too, though zingtouch seems pretty complete 👍

**Comment by rexagod** [comment]
@sashadev-sky Are you working on this at the moment?

**Comment by sashadev-sky** [comment]
@rexagod Hi! So everything that you see I moved into "next up" pipeline means I intend to work on it in the next month or so. I think I marked about 5 issues in LDI with this pipeline. 

@jywarren and I discussed issues I should work on and marked these. But that does not mean I am claiming them for myself! If you want to do one go ahead! Just write you're grabbing it. There's always more things to work on :) I am currently going to be working on exporter for mapknitter as soon as I close toolbar positioning and we get your abstract UI reviewed

**Comment by sashadev-sky** [comment]
@rexagod Can you see my ZenHub or is that just me? For me it's a tab at the top next to pull requests. It would be cool if you used it with me! And made a column for yourself too so we can keep up to date about eachother's in progress and goals, etc. easily.

and @jywarren if you can see my Zenhub too you can make a pipeline for both rexagod and I to grab issues out of. Although there are not so many that it's hard to see what to work on and we already know which ones broadly you are focused on for next LDI release / exporter. 

Lowkey determined to get all of them closed though!! 💥

**Comment by sashadev-sky** [comment]
@rexagod also sorry I didn't review your new work today. Needed a day to rest. Just responding to things before bed. Handling it tomorrow. :)

**Comment by rexagod** [comment]
@sashadev-sky I've set up the ZenHub app from the GitHub marketplace and now things are just like you described them (ZenHub tab, pipelines, etc.). Regarding the issues, I agree to have a semantic distribution of them between us, although, the question now is, do I assign issues to me on my own or is @jywarren interested in doing so for the both of us? Thanks!

**Comment by sashadev-sky** [comment]
@rexagod ah so cool so you can see my pipelines and setup? can you make a pranshu column?? or rexagod whatever you want to call it 😅

**Comment by sashadev-sky** [comment]
ZenHub has been a game-changer for me. I can see every issue so easily and organize them all (export, toolbar, images, etc) pipelines. Connect them to other issues and create Epics which are issues that will then be broken down on a large set of issues. And the Epic will have a progress tracker for how many of the sub issues are complete.

**Comment by rexagod** [comment]
@sashadev-sky Thanks for letting me know about this, I'll make a column for myself. I guess this will definitely have some productive outcomes.

**Comment by sashadev-sky** [comment]
@rexagod yess! I don't see any columns from you, have you tried it yet? Just want to make sure us sharing one works

**Comment by rexagod** [comment]
@sashadev-sky I just tried to make one, and apparently I do not have the permissions to create a pipeline. A little help here? Thanks!

**Comment by sashadev-sky** [comment]
Oh, Jywarren has to add permissions for you to do this (don't know what level of permission this is but he knows!) @jywarren can you please add for rexagod so we can better collaborate :) 

One thing about ZenHub - be careful with it - the "closed" column is not just a semantic name, moving something there will actually close the issue or PR. and each time you move pipelines it will create a notification on the page so try not to move too much (I already do this too much and feel bad 😅)

**Comment by jywarren** [comment]
Oh permissions in GitHub or zenhub? Cool!

**Comment by sashadev-sky** [comment]
It would be permissions for this repo in github. Not sure what you gave me but I guess if you can see which ones I have the same for rexagod? I don't believe it is a high permission level

**Comment by jywarren** [comment]
Ok, done!

**Comment by sashadev-sky** [comment]
@jywarren I was getting started on this and I was just wondering, for clarification, what is the purpose of the feature?

 Would it be to allow for rotation support regardless of handles, and then we get rid the `rotate` handles altogether? Because currently the image can already be rotated by dragging its handles in rotate mode. 

Or is it just to provide additional rotate  support in rotate mode, (i.e the user can drag any part of the image not just the handles)?

**Comment by jywarren** [comment]
Hi! I think this is just one extra way to do rotation/scaling, not instead
of the handles but in addition to them, because on multi-touch devices it'd
be quite intuitive. I'm sorry i had to run from the meeting so suddenly by
the way, but I was going to say that I understand that the positioning via
CSS transform is making this tough. It's not worth abandoning that and just
doing distortion but not positioning, is it? Does CSS transform result in
smoother panning of the image when dragging?

**Comment by sashadev-sky** [comment]
@jywarren what do you mean by this question: It's not worth abandoning that and just
doing distortion but not positioning, is it?

**Comment by jywarren** [comment]
Sorry, I wasn't super clear -- I was wondering if it would be easier to
stop positioning with css transform, and position in a more normal Leaflet
way, in order to make it easier to do the rotation implementation. But it's
probably not easier.

**Comment by sashadev-sky** [comment]
@jywarren I have something almost working down I think it could work I just need a little guidance on how to fix it.. Ill upload it soon!

**Comment by sashadev-sky** [comment]
Update: (finally) figured out that this can be done by extending rotateScale handlers, making them invisible and always on, and setting them to a size a bit larger than the image. Have not figured out how to make the handle size dynamic, so that when the user makes the image bigger the handle becomes bigger too. @rexagod any suggestions?
